test/test_datapipe.py

```diff
@@ -3145,12 +3126,15 @@ def test_simple_snapshot_graph_with_serialization(self):
         res2 = [10 * x for x in res1]
         self._snapshot_test_helper(graph2, expected_res=res2)
 
-        rng = torch.Generator()
-        graph3 = graph2.shuffle()
-        rng.manual_seed(0)
-        torch.utils.data.graph_settings.apply_shuffle_seed(graph3, rng)
-        res3 = list(graph3)
-        self._snapshot_test_helper(graph3, expected_res=res3)
+        # TODO: shuffle is not simple graph anymore
+        graph3 = graph2
+        res3 = res2
+        #  rng = torch.Generator()
+        #  graph3 = graph2.shuffle()
+        #  rng.manual_seed(0)
+        #  torch.utils.data.graph_settings.apply_shuffle_seed(graph3, rng)
+        #  res3 = list(graph3)
+        #  self._snapshot_test_helper(graph3, expected_res=res3)
```

Comment:
After changing the state after `serialization`, `shuffler` becomes non-simple graph. cc: @NivekT

Reply:
It becomes a non-simple graph as in there is a loop in the graph?

Actually I looked into this and I don't think that is the case. See my other comment

Reply:
I am changing this PR without touching the state of `Shuffler`. So, `shuffler` is still working properly with simple snapshot by always `reset`.